static/app/views/alerts/rules/issue/messagingIntegrationModal.tsx

```diff
@@ -0,0 +1,66 @@
+import {Fragment, useEffect, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {addErrorMessage} from 'sentry/actionCreators/indicator';
+import type {ModalRenderProps} from 'sentry/actionCreators/modal';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import type {Project} from 'sentry/types';
+import type {Organization} from 'sentry/types/organization';
+import AddIntegrationRow from 'sentry/views/alerts/rules/issue/addIntegrationRow';
+
+type Props = ModalRenderProps & {
+  organization: Organization;
+  project: Project;
+};
+
+function MessagingIntegrationModal({
+  closeModal,
+  Header,
+  Body,
+  organization,
+  project,
+}: Props) {
+  const [hasError, setHasError] = useState(false);
+  const integrationValues = ['slack', 'discord', 'msteams'];
+
+  useEffect(() => {
+    if (hasError) {
+      closeModal();
+      addErrorMessage(t('Failed to load integration data'));
+    }
+  }, [hasError, closeModal]);
```

Comment:
If the modal content can't be loaded, the modal is closed and a small error message appears in the corner:

https://github.com/user-attachments/assets/621725cd-8ffd-4c91-af0c-8e91483bfcaa

Reply:
hmm, the ux of this feels a little awkward. i wonder if we could block opening the popup until we've confirmed the integration can load instead? that way the UX would just show the toast instead of flashing the popup momentarily.

to do that, maybe have the button click trigger the xhr request, then on success have the modal open. This introduces a new problem of a "slow to open" modal, so maybe changing the staet of `connect to messaging` to a loading spinner while the xhr request is open? (should be able to get all the state of the request from react-query state)